### ISSUE TITLE
Generalize AccountsRawContactsQuery to RawContactsQuery

### ISSUE THREAD

**Issue by vestrel00** [vestrel00/contacts-android#271]
## Problem

Due to the changes in #268 and #269, the purpose and design of the current `AccountsRawContactsQuery` no longer makes sense.

## Solution

Generalize and expand the functionality of `AccountsRawContactsQuery` to `RawContactsQuery`. This is pretty much like `Query` except `RawContact` entities are returned instead of `Contact` entities.

- [x] Refactor `AccountsRawContactsQuery` to `RawContactsQuery`
- [x] Update `RawContactRefresh.kt` to use `RawContactsQuery`
- [x] Update `DataRawContact.kt` to use `RawContactsQuery`
- [x] Implement `include`s functions

Additionally;

- [x] Update sample app
- [x] Update cheatsheet (also move to basics section)
- [x] Update relevant documentation in code
- [x] Update relevant documentation in docs (also rename and move to basics section)
- [x] Verify that there are no regressions for related API functions
- [ ] Provide a migration guide as a comment in this issue so that it can be copy-pasted into the release notes

**Comment by vestrel00** [comment]
## Migration guide (to be included in release notes)

Previously, the `AccountsRawContactsQuery` had a very limited set of functions; https://github.com/vestrel00/contacts-android/blob/0.2.4/docs/accounts/query-raw-contacts.md

Its sole purpose was to provide a way to get RawContacts directly via the `BlankRawContact` entity. It did not allow you to specify fields to include as it had a predefined set of included fields.

You may have used it like so,

```kotlin
val rawContacts = Contacts(context).accounts().queryRawContacts().find()
```

The equivalent, yet more powerful, usage now is,

```kotlin
val rawContacts = Contacts(context).rawContactsQuery().find()
```

For full API docs, visit https://vestrel00.github.io/contacts-android/basics/query-raw-contacts/

**Comment by vestrel00** [comment]
This is included in [0.3.0](https://github.com/vestrel00/contacts-android/releases/tag/0.3.0)!

**Comment by vestrel00** [comment]
This is included in [0.3.0](https://github.com/vestrel00/contacts-android/releases/tag/0.3.0)!
